### ISSUE TITLE
Slash Command /friend is broken

### ISSUE THREAD

**Issue by fang2hou** [Stanzilla/WoWUIBugs#53]
The slash command `/friend` is broken.
```
1x FrameXML\ChatFrame.lua:2133: Usage: C_FriendList.AddOrRemoveFriend(name, notes)
[string "=[C]"]: in function `AddOrRemoveFriend'
[string "@FrameXML\ChatFrame.lua"]:2133: in function `?'
[string "@FrameXML\ChatFrame.lua"]:5073: in function `ChatEdit_ParseText'
[string "@FrameXML\ChatFrame.lua"]:4737: in function `ChatEdit_SendText'
[string "@FrameXML\ChatFrame.lua"]:4773: in function <FrameXML\ChatFrame.lua:4769>
[string "=[C]"]: in function `ChatEdit_OnEnterPressed'
[string "*:OnEnterPressed"]:1: in function <[string "*:OnEnterPressed"]:1>

Locals:
(*temporary) = nil
(*temporary) = nil
```

**Comment by Ketho** [comment]
You mean if you're targeting a player? The wiki [mentions](https://wow.gamepedia.com/MACRO_friend) _"If you do not specify who to add to your friends list, and are targeting a player, this command will add that player."_

I can reproduce this error on live and classic

**Comment by fang2hou** [comment]
Even if I am not targeting a player, it will also throw the error.
The path I found this bug is that I try to build a button with macrotext `/friend` to open Friend frame in combat.

**Comment by fang2hou** [comment]
Btw, when I try to use macro `/click CharacterMicroButton` to open character panel, it always failed. I also tried to add the button name and add `true` for simulating a down-click.

**Comment by Ketho** [comment]
About showing/holding panels in combat, that might be an intentional change
https://github.com/Gethe/wow-ui-source/blob/9.0.1/FrameXML/UIParent.lua#L3426

**Comment by Meorawr** [comment]
I believe `/friend` was fixed in 9.0.2.36665. Won't be fixed on Classic of course, but then again nothing ever will :)

**Comment by Ketho** [comment]
I don't know how it was fixed because I can't see it in the diff, but I can confirm this.

It also works in combat, although the first time it fails and any subsequent tries work. Classic doesn't exist so I'll close it x)
